homeassistant/components/mqtt/fan.py

```diff
@@ -321,12 +336,19 @@ def _setup_from_config(self, config):
         if self._feature_preset_mode:
             self._supported_features |= SUPPORT_PRESET_MODE
 
-        for key, tpl in list(self._templates.items()):
+        for key, tpl in list(self._value_templates.items()):
+            if tpl is None:
+                self._value_templates[key] = lambda value: value
+            else:
+                tpl.hass = self.hass
+                self._value_templates[key] = tpl.async_render_with_possible_json_value
+
+        for key, tpl in list(self._command_templates.items()):
```

Comment:
Instead of duplicating the code, maybe something like this:
```suggestion
        for tpl_dict in [self._command_templates, self._value_templates]
            for key, tpl in tpl_dict.items():
                ...
```